Sources/Algorithms/Intersperse.swift

```diff
@@ -12,8 +12,12 @@
 /// A sequence that presents the elements of a base sequence of elements
 /// with a separator between each of those elements.
 public struct Intersperse<Base: Sequence> {
-  let base: Base
-  let separator: Base.Element
+  @usableFromInline
+  internal let base: Base
+  
+  @usableFromInline
+  internal let separator: Base.Element
```

Comment:
Does this _need_ to be made internal? Is there any benefit to making it visible?

Reply:
This is a really good question, the only problem I can think is where `Base.Element` is a reference or we make `set` public, exposing it may allow users to modify state causing the wrapper collection to change behavior, but I'm not sure if this is really a problem ...
 
> Is there any benefit to making it visible?
I can't think of an use case where someone could need access to this property.

Reply:
You can't set a stored `let`.

Reply:
Ah! indeed you right, so the only concern would be references then e.g. 
```
struct CollectionWrapper<Base> where Base: Collection {
  let base: Base
  let separator: Base.Element
}

class Ref: Equatable {
  var b: Bool = true
  
  init() {}
  init(_ b: Bool) { self.b = b }
  
  static func == (lhs: Ref, rhs: Ref) -> Bool {
    return lhs.b == rhs.b
  }
}

let l: [Ref] = [Ref(), Ref(false), Ref()]

var c = CollectionWrapper(base: l, separator: Ref(false))

print(c.separator.b)

c.separator.b = true
// Any operation on `c` after that may have behavior changed 

print(c.separator.b) 
```

Although is very specific, this would be possible if we have the public property and can lead to unexpected behavior
But I still not sure if we should consider this really a problem here ... WDYT?

Reply:
I'm not sure I see the problem. The same behavior would happen if the reference were mutated not via `c.separator`: that's just what it means to have a reference type.

Reply:
> I'm not sure I see the problem. The same behavior would happen if the reference were mutated not via c.separator: that's just what it means to have a reference type.

Ah, indeed! You are totally right :) 

> Overall, while I don't have a concrete use case in mind, I do rather think it's reasonable for a user to want to know what separator they are using here, and to query that through the type that's using the separator. It is a property that's taken into account as part of the type's equivalence relation, is it not?

I agree this is reasonable and there isn't really a real reason to make it internal, so I think we could change.

Reply:
> Overall, while I don't have a concrete use case in mind, I do rather think it's reasonable for a user to want to know what separator they are using here, and to query that through the type that's using the separator.

@xwu Would you say the same argument could be made when deciding whether to make the `base` collection private, and if not, what makes it different?

Reply:
The rationale for removing the public accessibility of `base` is given in [issue 84](https://github.com/apple/swift-algorithms/issues/84).

Reply:
Sure, I was just curious whether you thought your reasoning would also extend to the `base` property if we ignore the concern about substitutability. Do you think it's reasonable for a user to want to access the base collection?

Reply:
I was thinking 2 things this after @timvermeulen comments 

* Does it make sense to be able to access what separator or size the wrapper collection is using if they can't access the base collection? 
* I was wondering that for the user perspective the most important thing is that this is a collection that behave as expected, kind of in the same sense that an eager implementation that just produces a result array. So by this reasoning specific properties of the wrapper type and I'm not sure but maybe even the wrapper type itself are kind of transparent for the user in the sense that they probably won't even expect to have access to those kind of specific collection properties. This is more a question than an affirmation, I'm still thinking about it... WDYT? @xwu @timvermeulen

Reply:
> Sure, I was just curious whether you thought your reasoning would also extend to the base property if we ignore the concern about substitutability. Do you think it's reasonable for a user to want to access the base collection?

My _only_ concern is about substitutability.